### PR TITLE
Added order table and implementation on post checkout

### PR DESCRIPTION
**Pull request by yankai14**

### CODE REVIEW

File: be/api/v1/endpoints/cart/checkout/post.py [hunk 1]
```diff
@@ -48,8 +49,28 @@ async def post_checkout(req: CheckoutRequestBodyModel):
             # setup_future_usage="on_session", # cannot be used with paynow
         )
 
+        orderID = uuid.uuid4().__str__()
+        orderDateTime = datetime.now().__str__()
+        customerEmail = req.email
+        transactionID = "payment_intent.id"
```
Comment (chanbakjsd) [inline be/api/v1/endpoints/cart/checkout/post.py:55]: Payment intent ID is not being stored correctly.

File: utils/dal/order.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+from utils.aws.dynamodb import read_item_from_db, write_item_to_db
+
+table_name = os.environ["ORDERS_TABLE_NAME"]
+
+def dal_create_order(order: Order):
+    orderItems = [{
+        "id": orderItem.id,
+        "name": orderItem.name,
+        "price": int(orderItem.price*100),
+        "images": orderItem.images,
```
Comment (chanbakjsd) [inline utils/dal/order.py:12]: Is it necessary to store images in an order?

File: utils/dal/order.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    } for orderItem in order.orderItems]
+
+    item = {
+        "orderID": order.orderID,
+        "orderDateTime": order.orderDateTime.__str__(),
+        "customerEmail": order.customerEmail,
+        "transactionID": order.transactionID,
+        "paymentGateway": order.paymentGateway,
+        "status": order.status.value,
+        "orderItems": orderItems
```
Comment (chanbakjsd) [inline utils/dal/order.py:26]:
```suggestion
        "orderItems": orderItems,
```

File: utils/dal/order.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+
+    write_item_to_db(table_name, item)
+
+    print(read_item_from_db(table_name, {"orderID": order.orderID}))
```
Comment (chanbakjsd) [inline utils/dal/order.py:32]: Is this `print` necessary?

**Review by chanbakjsd — CHANGES_REQUESTED** [review]

File: serverless.yml [hunk 5]
```diff
@@ -215,3 +217,17 @@ resources:
...
+      Properties:
+        TableName: ${self:provider.environment.ORDERS_TABLE_NAME}
+        AttributeDefinitions:
+          - AttributeName: orderID
+            AttributeType: S
+        KeySchema:
+          - AttributeName: orderID
+            KeyType: HASH
+        ProvisionedThroughput:
+          ReadCapacityUnits: 1
```
Comment (aryans1204) [inline serverless.yml:232]: Might want to change read and write capacities to 2 units, since transactional read/writes consume 2 units.

**Review by aryans1204 — CHANGES_REQUESTED** [review]
Dynamo DB stuff looks ok, though I noticed the payment intent ID is not stored correctly, as mentioned by @chanbakjsd .

**Review by chanbakjsd — APPROVED** [review]

**Review by aryans1204 — APPROVED** [review]
LGTM 🚀

**Pull request merged**